xmtp/src/storage/encrypted_store/mod.rs

```diff
@@ -161,35 +163,56 @@ impl EncryptedMessageStore {
         Ok(account_list.pop())
     }
 
-    pub fn get_session_with_conn(
+    pub fn get_latest_session_for_installation(
         &self,
         installation_id: &str,
         conn: &mut DbConnection,
     ) -> Result<Option<StoredSession>, StorageError> {
         use self::schema::sessions::dsl::*;
 
-        let mut session_list = sessions
+        sessions
             .filter(peer_installation_id.eq(installation_id))
-            .order(created_at.desc())
-            .load::<StoredSession>(conn)
-            .map_err(|e| StorageError::Unknown(e.to_string()))?;
+            .order(updated_at.desc())
+            .first(conn)
+            .optional()
+            .map_err(|e| StorageError::Unknown(e.to_string()))
+    }
 
-        warn_length(&session_list, "StoredSession", 1);
-        Ok(session_list.pop())
+    pub fn get_latest_sessions_for_installation(
+        &self,
+        installation_id: &str,
+        conn: &mut DbConnection,
+    ) -> Result<Vec<StoredSession>, StorageError> {
+        use self::schema::sessions::dsl::*;
+
+        sessions
+            .filter(peer_installation_id.eq(installation_id))
+            .order(updated_at.desc())
+            .get_results(conn)
+            .map_err(|e| StorageError::Unknown(e.to_string()))
     }
 
-    pub fn get_sessions(
+    pub fn get_latest_sessions(
         &self,
         user_address: &str,
         conn: &mut DbConnection,
     ) -> Result<Vec<StoredSession>, StorageError> {
-        use self::schema::sessions::dsl as schema;
+        if !is_wallet_address(user_address) {
+            return Err(StorageError::Unknown(
+                "incorrectly formatted walletAddress".into(),
+            ));
+        }
+        let session_list = sql_query("Select sessions.* from
+                    (
+                        select distinct
+                            first_value(session_id) over (partition by peer_installation_id order by updated_at desc) as session_id
+                        from sessions
+                        where user_address = ?
+
+                    )as ids
+                left join sessions on ids.session_id = sessions.session_id
+        ").bind::<Text,_>(user_address).load::<StoredSession>(conn).map_err(|e| StorageError::Unknown(e.to_string()))?;
```

Comment:
Would love to use diesel here, but couldn't find another way. Minimally the inputs are checked for length, and characters for added safety.

An approach that tags sessions as 'active' could be looked at but has more opportunity for bugs in managing the state